### PR TITLE
Make apiweb redirect pages no-auth

### PR DESCRIPTION
**Pull request by purplexa**

### CODE REVIEW

File: api_server/modules/web/newrelic_ide_redirect_request.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		let autoOpen = !!(!queryStringIDE || queryStringIDE === 'default');
+
+		if (queryStringIDE && queryStringIDE !== 'default') {
+			const mappedQueryStringIDE = ides.find(_ => _.moniker === queryStringIDE);
+			if (mappedQueryStringIDE) {
+				return { ides: ides, lastOrigin: mappedQueryStringIDE, autoOpen: true, isSpecificIde: true };
+			}
+		}
+
+		const lastOrigin = ((function() {
```
Comment (eric-j) [inline api_server/modules/web/newrelic_ide_redirect_request.js:54]: Could you add a comment here explaining what this code is doing/how we are using cookies to determine lastOrigin?   More for my sake if I ever end up working with this code.

**Review by eric-j — APPROVED** [review]
👍 Good stuff

**Review by cstryker — APPROVED** [review]

**Review by CalvinAllen — APPROVED** [review]
Just a few questions, but otherwise looks good

File: api_server/modules/web/codemark_ide_redirect_request.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	getTeamId () {
+		return this.decodeLinkId(this.request.params.teamId);
+	}
+
+	async getCodemarkId (teamId) {
+		const linkId = this.decodeLinkId(this.request.params.id, 2);
+		const codemarkLink = await this.data.codemarkLinks.getOneByQuery(
+			{ teamId: teamId, _id: linkId },
+			{ hint: CodemarkLinkIndexes.byTeamId }
+		);
```
[the remark this answers is not in the document]
Comment (cstryker) [reply api_server/modules/web/codemark_ide_redirect_request.js:30]: @CalvinAllen is correct ... if querying by id, just call `this.data.codemarkLinks.getById(linkId)`. No hint.

File: api_server/modules/web/codeerror_ide_redirect_request.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	getTeamId () {
+		return this.decodeLinkId(this.request.params.teamId);
+	}
+
+	async getCodeErrorId (teamId) {
+		const linkId = this.decodeLinkId(this.request.params.id, 2);
+		const codemarkLink = await this.data.codemarkLinks.getOneByQuery(
+			{ teamId: teamId, _id: linkId },
+			{ hint: CodemarkLinkIndexes.byTeamId }
+		);
```
[the remark this answers is not in the document]
Comment (cstryker) [reply api_server/modules/web/codeerror_ide_redirect_request.js:30]: same answer :)

File: api_server/modules/web/review_ide_redirect_request.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+	getTeamId () {
+		return this.decodeLinkId(this.request.params.teamId);
+	}
+
+	async getReviewId (teamId) {
+		const linkId = this.decodeLinkId(this.request.params.id, 2);
+		const codemarkLink = await this.data.codemarkLinks.getOneByQuery(
+			{ teamId: teamId, _id: linkId },
+			{ hint: CodemarkLinkIndexes.byTeamId }
+		);
```
[the remark this answers is not in the document]
Comment (cstryker) [reply api_server/modules/web/review_ide_redirect_request.js:29]: same :)

**Comment by cstryker** [comment]
good catches @CalvinAllen

**Pull request merged**